packages/ember-data/lib/system/record-array-manager.js

```diff
@@ -153,7 +153,7 @@ export default Ember.Object.extend({
     for (var i = 0, l = records.length; i < l; i++) {
       record = records[i];
 
-      if (!get(record, 'isDeleted') && !get(record, 'isEmpty')) {
+      if (!record.isDeleted() && !record.isEmpty()) {
```

Comment:
It won't because record here is in fact a reference. Anything you touch internally is a reference. I just need to go and rename everything once we have a name.